model/connect/MySQLiConnector.php

```diff
@@ -235,18 +238,16 @@ public function preparedQuery($sql, $parameters, $errorLevel = E_USER_ERROR) {
 			$statement->execute();
 			return $statement;
 		});
-
-		// check result
-		$this->setLastStatement($lastStatement);
+		
 		if (!$lastStatement || $lastStatement->error) {
 			$values = $this->parameterValues($parameters);
 			$this->databaseError($this->getLastError(), $errorLevel, $sql, $values);
 			return null;
 		}
 
-		// May not return result for non-select statements
-		if($result = $lastStatement->get_result()) {
-			return new MySQLQuery($this, $result, $lastStatement);
+		// Non-select queries will have no result data
+		if($lastStatement && ($metaData = $lastStatement->result_metadata())) {
```

Comment:
Discussed with Damian; we're going to return a stub MySQLQuery just like 3.1 does.